[DMS-76] Persistent ordered map unit testing

s-and-witch opened the pull request:
In addition to tests, this patch fixes issues when `foldRight` actually does a left folding and vice versa.

GoPavel reviewed (approved):
I would suggest some changes:
+ add unit test of `replace` on empty map
+ test `iter` for more than 1 element (as for entries)
+ test `size` for more than 1 element

Anyway I am approving in advance

GoPavel commented:
>In addition to tests, this patch fixes issues when foldRight actually does a left folding and vice versa.

Wait, actually mistake in the tests.
You have: 
```
test(
      "left fold values",
      Map.foldLeft(buildTestMap(), "", concatenateValues),
      M.equals(T.text("210"))
    ),
```
But 
```
foldLeft([1, 2, 3], "", f) === f (3, f (2, f (1, ""))) === "" # 1 # 2 # 3 == 123
where f = \k, v, acc -> acc #  v
```
I think we should revert the fix and fix the tests.

s-and-witch commented:
@GoPavel good catch! 
I applied your suggestions:
- Reverted changes related to folds and fix tests
- Removed `direction` argument from `keys` and `vals` functions
- Added a new function `empty`.
- Added new tests for replace, iter and size functions
